### PR TITLE
Fix stateless inference for transformer-only pipelines.

### PR DESCRIPTION
**Pull request by ColdTeapot273K**
Addresses #1269

### CODE REVIEW

File: river/compose/pipeline.py [hunk 1]
```diff
@@ -568,13 +567,17 @@ def _transform_one(self, x: dict):
             if utils.inspect.ischildobject(obj=t, class_name="AnomalyFilter"):
                 continue
 
+            print(f"{t=}")
```
Comment (MaxHalford) [inline river/compose/pipeline.py:570]: Can you remove `print` statements?

File: river/compose/pipeline.py [hunk 2]
```diff
@@ -590,10 +593,14 @@ def transform_one(self, x: dict, **params):
...
 
         """
+        print(f"{self._STATELESS=}")
         x, last_step = self._transform_one(x)
+
         if isinstance(last_step, base.Transformer):
-            if not last_step._supervised:
-                last_step.learn_one(x)
+            if not self._STATELESS:
+                if not last_step._supervised:
```
Comment (MaxHalford) [inline river/compose/pipeline.py:601]:
```suggestion
            if not self._STATELESS and not last_step._supervised:
```

no?

File: river/compose/pipeline.py [hunk 3]
```diff
@@ -815,8 +822,9 @@ def transform_many(self, X: pd.DataFrame):
         """
         X, last_step = self._transform_many(X=X)
         if isinstance(last_step, base.MiniBatchTransformer):
-            if not last_step._supervised:
-                last_step.learn_many(X)
+            if not self._STATELESS:
+                if not last_step._supervised:
```
Comment (MaxHalford) [inline river/compose/pipeline.py:826]:
```suggestion
            if not self._STATELESS and not last_step._supervised:
```

**Review by MaxHalford — CHANGES_REQUESTED** [review]
Great catch! Could you add a unit test to test this logic?

File: river/compose/pipeline.py (hunk 2 above)
Comment (ColdTeapot273K) [reply river/compose/pipeline.py:601]: Sure, just wanted to keep it in line with how things a written in other methods :)

**Comment by MaxHalford** [comment]
If it's ok I'll close this, as I'm integrating the same modification in #1277

**Pull request closed without merging**